alacritty/src/display/mod.rs

```diff
@@ -1125,19 +1130,22 @@ impl Display {
         rects.extend(underline.rects(Flags::UNDERLINE, &metrics, &self.size_info));
 
         let ime_popup_point = match preedit.cursor_end_offset {
-            Some(cursor_end_offset) if cursor_end_offset != 0 => {
-                let is_wide = preedit.text[preedit.cursor_byte_offset.unwrap_or_default()..]
-                    .chars()
-                    .next()
-                    .map(|ch| ch.width() == Some(2))
-                    .unwrap_or_default();
+            Some(cursor_end_offset) => {
+                // Regular edit happens when the start == end, thus use Beam for it, however
+                // when multiple characters are changed at once use the `HollowBlock`.
+                let (shape, width) = if let Some(width) =
+                    NonZeroU32::new((cursor_end_offset.0 - cursor_end_offset.1) as u32)
+                {
+                    (CursorShape::HollowBlock, width)
```

Comment:
This uses hollow block for any width 2 characters, right? Shouldn't that still use beam cursor?

Reply:
The way it's written is correct, since single character is a width of 0, since cursors point to the same offset within the text. If you have non zero distance between offsets it means that it covers more than one character.

As you can see on the screenshot it works as expected.